server/src/index.ts

```diff
@@ -1,38 +1,45 @@
-import express from 'express';
-import mysql from 'mysql2';
+import express from 'express'
+import mysql from 'mysql2'
 
-import Config from './config';
-import logMiddleware from './middleware/logger';
-import errMiddleware from './middleware/errors';
+import Config from './config'
+import logMiddleware from './middleware/logger'
+import errMiddleware from './middleware/errors'
 
-import HealthRouter from './api/health/router';
+import HealthRouter from './api/health/router'
 
-import { createFeatureRouter } from './api/feature/router';
+import { createFeatureRouter } from './api/feature/router'
+import { createUserRouter } from './api/users/router'
 
-const app = express();
+const app = express()
 
 const BASE_PATH = '/api/v1'
 
 async function main() {
-  const config = Config.readFromEnvironment();
-  const db = mysql.createConnection(config.dbOptions());
-  const healthRouter = new HealthRouter(db);
-  const featureRouter = createFeatureRouter(db);
+  const config = Config.readFromEnvironment()
+  const db = mysql.createConnection(config.dbOptions())
+  const healthRouter = new HealthRouter(db)
+  const featureRouter = createFeatureRouter(db)
+  const userRouter = createUserRouter(db)
 
   // apply middleware
-  app.use(express.json());
-  app.use(logMiddleware);
+  app.use(express.json())
+  app.use(logMiddleware)
 
   // setup routers
-  app.use(BASE_PATH, healthRouter.routes());
-  app.use(BASE_PATH, featureRouter.routes());
+  app.use(BASE_PATH, healthRouter.routes())
```

Comment:
You could set up another router that provides the base path as well.

Reply:
No worries at all, It'll make thing easier to read, but also helps with pattern matching. For example:
```
// in birds.js
var express = require('express')
var router = express.Router()

// middleware that is specific to this router
router.use(function timeLog (req, res, next) {
  console.log('Time: ', Date.now())
  next()
})
// define the home page route
router.get('/', function (req, res) {
  res.send('Birds home page')
})
// define the about route
router.get('/about', function (req, res) {
  res.send('About birds')
})

module.exports = router

// in base file
var birds = require('./birds')
app.use('/birds', birds)
```
The router is matched on a subset of paths the user can request to. You can also then just pass the whole router to use instead.

Reply:
I think for the sake of time, I'm gonna hold off of this restructure until the project is close to finished. Right now I'd like to focus on getting all merged together so I can start testing features. That sound good to you? I saved all the info here in my notes as well.